src/qibo/models/encodings.py

```diff
@@ -558,3 +659,166 @@ def _parametrized_two_qubit_gate(gate, q0, q1, params=None):
         return gate(q0, q1, *params)
 
     return gate(q0, q1)
+
+
+def _angle_mod_two_pi(angle):
+    return angle % (2 * np.pi)
+
+
+def _get_markers(bitstring, last_run: bool = False):
+    nqubits = len(bitstring)
+    markers = [len(bitstring) - 1]
+    for ind, value in zip(range(nqubits - 2, -1, -1), bitstring[::-1][1:]):
+        if value == bitstring[-1]:
+            markers.append(ind)
+        else:
+            break
```

Comment:
Also here not 100% sure is equivalent, but if I got it right you are checking whether there is another element (except the last one bitstring[-1]) in bitstring with the same value as bitstring[-1], right?
```suggestion
    ind = bitstring[::-1][1:].index(bitstring[-1])
    markers.append(
        list(range(nqubits - 2, -1, -1))[ind]
    )
```